tests/integration/file_convert_test.go

```diff
@@ -72,3 +73,69 @@ func Test_FileConvert(t *testing.T) {
 		})
 	}
 }
+
+func Test_FileConvert_28xTo34x(t *testing.T) {
+	originalCwd, err := os.Getwd()
+	require.NoError(t, err)
+	defer func() {
+		// Restore the original working directory after the test
+		err := os.Chdir(originalCwd)
+		require.NoError(t, err)
+	}()
+
+	tests := []struct {
+		name               string
+		inputFile          string
+		expectedOutputFile string
+		errorExpected      bool
+		errorString        string
+	}{
+		{
+			name:               "auto-fixes plugin configuration for 2.8x",
+			inputFile:          "testdata/file-convert/002-kong-gateway-28x-to-34x-migration/28x-plugins.yaml",
+			errorExpected:      false,
+			expectedOutputFile: "testdata/file-convert/002-kong-gateway-28x-to-34x-migration/34x-expected-plugins.yaml",
+		},
+		{
+			name:               "auto-fixes route configuration for 2.8x",
+			inputFile:          "testdata/file-convert/002-kong-gateway-28x-to-34x-migration/28x-routes.yaml",
+			errorExpected:      false,
+			expectedOutputFile: "testdata/file-convert/002-kong-gateway-28x-to-34x-migration/34x-expected-routes.yaml",
```

Comment:
Lets add consumer and consumer group scoped plugins in the tests too?

Reply:
The plugins that we are auto-fixing right now aren't supported for consumer-group scoping as checked from here:
https://docs.konghq.com/hub/plugins/compatibility/#scopes

I have added a possible one for consumer-scoping. Also, expanded the tests for route-scoped and service+route scoped plugin.